ilc/common/trailingSlash.js

```diff
@@ -0,0 +1,31 @@
+const routerHasTo = {
+    doNothing: 'doNothing',
+    redirectToBaseUrl: 'redirectToBaseUrl',
+    redirectToBaseUrlWithTrailingSlash: 'redirectToBaseUrlWithTrailingSlash',
+};
+
+const processUrl = (trailingSlash, url) => {
+    const doesReqUrlEndWithTrailingSlash = url[url.length - 1] === '/';
```

Comment:
I believe you missed to handle case (and test for it) when URL has query params and/or hash.